RTClib.cpp

```diff
@@ -304,17 +342,46 @@ DateTime::DateTime (const __FlashStringHelper* date, const __FlashStringHelper*
     @param buffer: array of char for holding the format description and the formatted DateTime. 
                    Before calling this method, the buffer should be initialized by the user with 
                    a format string, e.g. "YYYY-MM-DD hh:mm:ss". The method will overwrite 
-                   the buffer with the formatted date and/or time.
+                   the buffer with the formatted date and/or time. Use the "AP" tag to
+                   return "AM" or "PM" and 12-Hour time.
     @return a pointer to the provided buffer. This is returned for convenience, 
             in order to enable idioms such as Serial.println(now.toString(buffer));
 */
 /**************************************************************************/
 
-char* DateTime::toString(char* buffer){
-		for(int i=0;i<strlen(buffer)-1;i++){
+char* DateTime::toString(char* buffer){        
+    char* apTag = strstr(buffer, "ap");
+    char* APTag = strstr(buffer, "AP");
```

Comment:
Since the actual location of the tag isn't used you could set a single flag here instead of checking both apTag and APTag later on. Something like `uint8_t apTag = ((strstr(buffer, "ap") != nullptr) || (strstr(buffer, "AP") != nullptr));`